### ISSUE TITLE
migrations fail when setting non-standard contracts_build_directory

### ISSUE THREAD

**Issue by szerintedmi** [trufflesuite/truffle#538]
- [x] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue
Migration fails when custom `contracts_build_directory` set in `truffle.js`
Issue already raised in [truffle-migrate project](https://github.com/trufflesuite/truffle-migrate/issues/10) but those issues might not be monitored

## Steps to Reproduce

1. set a non standard build directory in truffle.js (eg: `contracts_build_directory: './src/contractsBuild'`)
1. run `truffle migrate`

## Expected Behavior

Build should output to the directory `contracts_build_directory` set in `truffle.js`

## Actual Results
```
$ truffle migrate
Using network 'development'.

Running migration: 1_initial_migration.js
/usr/local/lib/node_modules/truffle/build/cli.bundled.js:49072
  throw new Error("Could not find artifacts for " + import_path + " from any sources");
  ^

Error: Could not find artifacts for ./Migrations.sol from any sources
    at Resolver.require (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:49072:9)
    at Object.require (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:66980:36)
    at ResolverIntercept.require (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:219957:32)
    at /Users/petro/ether/ucd-poc/migrations/1_initial_migration.js:1:28
    at ContextifyScript.Script.runInContext (vm.js:53:29)
    at ContextifyScript.Script.runInNewContext (vm.js:59:15)
    at /usr/local/lib/node_modules/truffle/build/cli.bundled.js:96391:14
    at tryToString (fs.js:513:3)
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:501:12)
```

## Environment

* Operating System: MacOs Sierra 10.12.6
* Truffle version: 3.4.9
* Ethereum client: N/A (using TestRPC 4.0.1) 
* node version: 8.2.1
* npm version:  5.3.0

**Comment by thisbejim** [comment]
bump

**Comment by CruzMolina** [comment]
Thanks for raising this issue @szerintedmi!

We've finally merged a fix for this into truffle@next via #1331.
